www/getChallenge.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+use SimpleSAML\Module\perun\DatabaseConnector;
+use SimpleSAML\Logger;
+
+$entityBody = file_get_contents('php://input');
+$body = json_decode($entityBody, true);
+
+$id = $body['id'];
+$scriptName = $body['scriptName'];
+
+const RANDOM_BYTES_LENGTH = 10;
+const TABLE_NAME = 'scriptChallenges';
+
+
+$challenge = hash('sha256', openssl_random_pseudo_bytes(RANDOM_BYTES_LENGTH));
+
+$databaseConnector = new DatabaseConnector();
+$conn = $databaseConnector->getConnection();
+
+assert($conn !== null);
+
+$stmt = $conn->prepare('INSERT INTO ' . TABLE_NAME . ' (id, challenge, script) VALUES (?, ?, ?)');
+
+if ($stmt) {
+    $stmt->bind_param('sss', $id, $challenge, $scriptName);
+    $ex = $stmt->execute();
+    if ($ex === false) {
+        Logger::error('Error while creating a challenge');
+    }
+
+    $stmt->close();
+} else {
+    Logger::error('Error during preparing statement');
```

Comment:
This should probably fail (with HTTP error code) and not continue. As it is, this script returns a challenge which was not saved (processing does not stop here), so the challenge will not work.